plugins/utils/dashboard.nix

```diff
@@ -6,141 +6,378 @@
   ...
 }:
 with lib;
-let
-  cfg = config.plugins.dashboard;
-in
-{
-  options = {
-    plugins.dashboard = {
-      enable = mkEnableOption "dashboard";
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "dashboard";
+  originalName = "dashboard-nvim";
+  defaultPackage = pkgs.vimPlugins.dashboard-nvim;
 
-      package = helpers.mkPluginPackageOption "dashboard" pkgs.vimPlugins.dashboard-nvim;
+  maintainers = [ helpers.maintainers.MattSturgeon ];
 
-      header = mkOption {
-        description = "Header text";
-        type = types.nullOr (types.listOf types.str);
-        default = null;
-      };
+  # TODO introduced 2024-05-30: remove 2024-09-01
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "dashboard"
+      ];
+    in
+    [
+      (mkRemovedOptionModule (
+        basePluginPath ++ [ "sessionDirectory" ]
+      ) "This plugin no longer has session support.")
+    ]
+    ++ (mapAttrsToList
+      (
+        old: new:
+        mkRenamedOptionModule (basePluginPath ++ [ old ]) (basePluginPath ++ [ "settings" ] ++ new)
+      )
+      {
+        header = [
+          "config"
+          "header"
+        ];
+        footer = [
+          "config"
+          "footer"
+        ];
+        center = [
+          "config"
+          "center"
+        ];
+        hideStatusline = [
+          "hide"
+          "statusline"
+        ];
+        hideTabline = [
+          "hide"
+          "tabline"
+        ];
+      }
+    )
+    ++ (mapAttrsToList
+      (
+        old: new:
+        mkRenamedOptionModule
+          (
+            basePluginPath
+            ++ [
+              "preview"
+              old
+            ]
+          )
+          (
+            basePluginPath
+            ++ [
+              "settings"
+              "preview"
+              new
+            ]
+          )
+      )
+      {
+        command = "command";
+        file = "file_path";
+        height = "file_height";
+        width = "file_width";
+      }
+    );
+
+  settingsOptions =
+    let
+      actionType = types.submodule {
+        freeformType = with types; attrsOf anything;
+        options = {
+          group = helpers.defaultNullOpts.mkStr "" ''
+            Highlight group used with the "hyper" theme, 
+          '';
+
+          icon = helpers.defaultNullOpts.mkStr "" ''
+            The icon displayed with this action.
+          '';
+
+          icon_hl = helpers.defaultNullOpts.mkStr "DashboardIcon" ''
```

Comment:
Some of these options are repeated in a few places. I think I'll define the directly in the `let` block and then inherit them where they're used.

Reply:
Yes, that is a good idea.

Reply:
While the options are _almost_ identical, the plugin-default varies and the description ought to slightly too.

Going to stick to having a few discrete definitions for now.